gazelle/deps.bzl

```diff
@@ -14,16 +14,9 @@
 
 "This file managed by `bazel run //:gazelle_update_repos`"
 
-load(
-    "@bazel_gazelle//:deps.bzl",
-    _go_repository = "go_repository",
-)
+load("@bazel_gazelle//:deps.bzl", "go_repository")
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
-def go_repository(name, **kwargs):
-    if name not in native.existing_rules():
-        _go_repository(name = name, **kwargs)
-
```

Comment:
Why are you removing this?

Reply:
Any idea what the original logic was there for? It looks like the intent was something similar to `maybe(http_archive, ...)` I usually see. Is this like a way that predates using `maybe`?

In any case, yeah, these "define repos for depenencies" things use a conditional definition so that the root workspace can override them. That intent should be kept.

Reply:
We've been carrying this patch for a while, and no-one on the team remembers why we removed this, but I agree it seems better to leave this as is. I've reverted that change.